src/GameObject.cpp

```diff
@@ -79,29 +79,11 @@ void GameObject::setYAxisRotation(float angle) {
 void GameObject::calculateAndSetInitialRotation() {

```

Comment:
Yeah, definitely another 'sore spot,' (unintentionally, I discovered the code is already suuuper brittle, as the Wall GameObject is affected w/o the if(velocity != 0.0) part).

However, I'm unsure of the best place for it. I'm hesitant to throw it in the render component as it's not rendering specific (e.g. it affects the bounding box transform as well).

Maybe just call it as part of the 'initObjectPhysics' call in the physics component? That way it's object specific and is just part of the normal init physics process if the object even needs it.